disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/serialization/mapping/logs/LogRecordDataMapper.java

```diff
@@ -45,7 +46,8 @@ public LogRecord mapToProto(LogRecordData source) {
       logRecord.body(bodyToAnyValue(source.getBody()));
     }
 
-    logRecord.flags(source.getSpanContext().getTraceFlags().asByte());
+    byte flags = source.getSpanContext().getTraceFlags().asByte();
+    logRecord.flags(toUnsignedInt(flags));
```

Comment:
is this comment in core java repo outdated?

https://github.com/open-telemetry/opentelemetry-java/blob/f123d789097d714ed94f76d98e87985067a7b301/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/LogMarshaler.java#L222-L225

Reply:
Your question relates to [this comment](https://github.com/open-telemetry/opentelemetry-java/pull/6167#discussion_r1466559349) from @jack-berg as well. This PR was motivated by that other one.

Reply:
oh, sorry, for some reason I thought you were calling the real `toUnsignedInt` here and missed the android-friendly implementation